### PR TITLE
Apps readme

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. tidy up README of apps
2. add new README to describe how to add new app

### CODE REVIEW

**Comment by sunghan-chang** [comment]
Markdown view is here.
https://github.com/sunghan-chang/TizenRT/blob/apps_readme/apps/HowtoAddNewApp.md

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/HowtoAddNewApp.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+# How to add new application
+
+Five files, **source file**, [**Kconfig**](#kconfig), [**Kconfig_ENTRY**](#kconfig_entry), [**Make.defs**](#makedefs) and [**Makefile**](#makefile) are mandatory to add a new application.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: I think it should be rephrased like this .
The following files are required to integrate a new application with the TizenRT build system :-
* Kconfig
* Kconfig_ENTRY
* Make.defs
* Makefile

File: apps/HowtoAddNewApp.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+## Kconfig
+This file contains configurations of new application to enable or to set specific value.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: This file is required by Kconfig-frontends menu build system. It is used to specify various config values according to which the application should be built. The following two config values must be defined in the Kconfig file.

File: apps/HowtoAddNewApp.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	bool "<Write Description>"
+	default n
+	---help---
+		<Write Details>
+
+config USER_ENTRYPOINT
+	string
+	default "<APP_MAIN_FUNCTION>" if ENTRY_<APP_NAME>
+```
+First controls enabling application and second provides a main function pointer for CONFIG_USER_ENTRYPOINT.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: The first config parameter specifies whether the application should be built or not.
The second config parameter specifies the function name of the application which should be executed when OS executes the application. The entrypoint can be set manually but for it the name of function which has to be executed firstly should be known for each application. This option registers the app with the build system and can be selected from the menu to be used as application entrypoint.

File: apps/HowtoAddNewApp.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+config USER_ENTRYPOINT
+	string
+	default "<APP_MAIN_FUNCTION>" if ENTRY_<APP_NAME>
+```
+The first config parameter specifies whether the application should be built or not.
+
+The second config parameter specifies the function name of the application which should be executed when OS executes the application. The entrypoint can be set manually but for it the name of function which has to be executed firstly should be known for each application. This option registers the app with the build system and can be selected from the menu to be used as application entrypoint.
+
+## Kconfig_ENTRY
+This contains only one configuration which allows entry point selection combined above USER_ENTRYPOINT configuration.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: selection *combined* above  -> selection *based on* above

File: apps/HowtoAddNewApp.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+```
+config ENTRY_<APP_NAME>
+	bool "<Write Description>"
+	depends on <APP_NAME>
+```
+
+Menuconfig seqeunce is shown below:
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: Steps in menuconfig to enable and selecting an app as userentry point  are shown below

File: apps/HowtoAddNewApp.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+```
+Application Configuration -> <APP_NAME> to y
+Application Configuration -> Application entry point -> <APP_NAME>
+```
+
+## Make.defs
+This shows the folder name of new application.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: This "specifies" .....

File: apps/HowtoAddNewApp.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+Menuconfig seqeunce is shown below:
+```
+Application Configuration -> <APP_NAME> to y
+Application Configuration -> Application entry point -> <APP_NAME>
+```
+
+## Make.defs
+This shows the folder name of new application.  
+Makefile of apps folder gets this information and includes in build when new application is enabled by configuration.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: Makefile of apps folder *reads this file and includes the folder in *build ....

File: apps/HowtoAddNewApp.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+else
+context:
+
+endif
+```
+Refer Builtin [README](builtin/README.md).
+
+#### Library information
+This shows that this application joins what library is and where it is.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: This specifies the library and its location into which the application would be built .

File: apps/HowtoAddNewApp.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+ifeq ($(WINTOOL),y)
+  BIN = ..\\libapps$(LIBEXT)
+else
+  BIN = ../libapps$(LIBEXT)
+endif
+endif
+```
+
+#### Caution
+If new application has many folder inside, Makefile of apps/examples/testcase is better to use.
```
Comment (shivgarg) [inline apps/HowtoAddNewApp.md]: If application has many folders inside, follow the Makefile of apps/example/testcase as an example.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/HowtoAddNewApp.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
+# How to add new application
```
Comment (btheosam) [inline apps/HowtoAddNewApp.md]: Which word would be apt here. "add" or "Create"?

File: apps/HowtoAddNewApp.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+else
+ifeq ($(WINTOOL),y)
+  BIN = ..\\libapps$(LIBEXT)
+else
+  BIN = ../libapps$(LIBEXT)
+endif
+endif
+```
+
+#### Caution
```
Comment (btheosam) [inline apps/HowtoAddNewApp.md]: Why is this a caution?

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**